xbmc/windowing/windows/WinSystemWin32DX.cpp

```diff
@@ -380,3 +381,41 @@ HRESULT APIENTRY HookOpenAdapter10_2(D3D10DDIARG_OPENADAPTER *pOpenData)
   }
   return hr;
 }
+
+bool CWinSystemWin32DX::IsHDRDisplay()
+{
+  if (CWIN32Util::GetWindowsHDRStatus() != HDR_STATUS::HDR_UNSUPPORTED)
+    return true;
+
+  return false;
+}
+
+HDR_STATUS CWinSystemWin32DX::GetOSHDRStatus()
+{
+  return CWIN32Util::GetWindowsHDRStatus();
+}
+
+HDR_STATUS CWinSystemWin32DX::ToggleHDR()
+{
+  return m_deviceResources->ToggleHDR();
+}
+
+bool CWinSystemWin32DX::IsHDRSupported() const
+{
+  return m_deviceResources->IsHDRSupported();
+}
+
+bool CWinSystemWin32DX::IsHDROutput() const
+{
+  return m_deviceResources->IsHDROutput();
```

Comment:
the same as for `CWinSystemWin10DX`. it looks like a copy of `GetOSHDRStatus` method

Reply:
Is not exactly the same:

`GetOSHDRStatus` obtains HDR capabilities from OS in realtime.
`IsHDROutput` is only cached status of swapchain configuration.

There could be a significant performance impact if `GetOSHDRStatus` is used all the time.

In general the sequence is:

1. `GetOSHDRStatus`
2. Configure swapchain with or witout HDR and sets  `IsHDROutput` flag.
3. Use `IsHDROutput` flag in the rest of calls e.g. RenderBase methods.
4. Toggle HDR when is necessary

Go to 1

Reply:
I had mixed things. you are right.